File: Java/BinarySearch.java
public class BinarySearch {
    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
        int target = 5;
        int left = 0;
        int right = arr.length - 1;
        int mid = (left + right) / 2;
        while (left <= right) {
            if (arr[mid] == target) {
                System.out.println("Found at index " + mid);
                break;
            } else if (arr[mid] < target) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
            mid = (left + right) / 2;
        }
        if (left > right) {
            System.out.println("Not found");
        }
    }
}

File: Python/binarySearch.py
import math
numbers = [1,2,4,3,7,5,6,8,9,10]
numbers.sort()
target = 5
def search(arr):
    low = 0
    high = len(arr) - 1
    mid = math.floor((low+high)/2)
    while low <= high:
        if arr[mid] == target:
            print("Found target")
            break
        elif arr[mid] < target:
            low = mid +1
            mid = math.floor((low+high)/2)
        elif arr[mid] > target:
            high = mid -1
            mid = math.floor((low+high)/2)
        else:
            print("Number Not Found")
search(numbers)